modules/tinymce/src/plugins/accordion/main/ts/core/Actions.ts

```diff
@@ -0,0 +1,52 @@
+import Editor from 'tinymce/core/api/Editor';
+
+import { fireInsertAccordionEvent } from '../api/Events';
+
+const nonRemovableContainers = [ 'BODY', 'TD', 'TH', 'LI', 'DIV', 'DT', 'DD' ];
```

Comment:
Not sure body should be in the valid set.

Reply:
`BODY` element can occur here as a `rng.commonAncestorContainer` when several paragraphs are selected. In this case I am inserting the `details` element at the end of the `BODY` element. Seems like the correct solution to me.


Without the `BODY` element in this list the `target` variable will be `null` and the accordion won't be inserted, which is a weird behavior.

Reply:
Right but not sure we should have the schema re-implemented in the accordion plugin. I think using the insert api is the right approach here since smif you say insert a table in the middle of a paragraph with text that gets split apart and the accordion is inserted the same thing should happen with the accordion inserting anywhere should just split the parent block and insert it in it's place. Since details is a block element and blocks can't contain blocks unless the block is a div or a td then it will split.

Reply:
Obsolete due to the latest changes.